service/frontend/workflow_handler.go

```diff
@@ -2418,12 +2415,10 @@ func (wh *WorkflowHandler) ListArchivedWorkflowExecutions(ctx context.Context, r
 		return nil, errRequestNotSet
 	}
 
+	maxPageSize := int32(wh.config.VisibilityArchivalQueryMaxPageSize())
 	if request.GetPageSize() <= 0 {
-		request.PageSize = int32(wh.config.VisibilityMaxPageSize(request.GetNamespace()))
-	}
-
-	maxPageSize := wh.config.VisibilityArchivalQueryMaxPageSize()
-	if int(request.GetPageSize()) > maxPageSize {
+		request.PageSize = maxPageSize
+	} else if request.GetPageSize() > maxPageSize {
 		return nil, serviceerror.NewInvalidArgument(fmt.Sprintf(errPageSizeTooBigMessage, maxPageSize))
```

Comment:
looks like for other APIs the logic have been changed to force override page size instead of returning invalid argument?

Reply:
For other API's, it returns error only if when it's using Elasticsearch and the page size is larger than `esIndexMaxResultWindow` (default 10000).
We decided to remove this config, and enforce the max page size to `visibilityMaxPageSize` (default 1000) without returning any error if the user uses a larger value.